usaspending_api/etl/management/helpers/load_submission.py

```diff
@@ -0,0 +1,166 @@
+            f"Unable to find object class for object_class={object_class}, direct_reimbursable={direct_reimbursable}."
+        )
+
+
+def get_object_class(row_object_class, row_direct_reimbursable):
+    """Lookup an object class record.
+
+        Args:
+            row_object_class: object class from the broker
+            row_direct_reimbursable: direct/reimbursable flag from the broker
+                (used only when the object_class is 3 digits instead of 4)
+    """
+
+    row = Bunch(object_class=row_object_class, by_direct_reimbursable_fun=row_direct_reimbursable)
+    return get_object_class_row(row)
+
+
+class CertifiedAwardFinancialIterator:
+    chunk_size = 100000
+
+    def __init__(self, submission_attributes):
+        self.submission_attributes = submission_attributes
+
+        # For managing state.
+        self._last_id = None
+        self._current_chunk = None
+
+    def _retrieve_and_prepare_next_chunk(self):
+        sql = f"""
+            select  *
+            from    certified_award_financial
+            where   submission_id = {self.submission_attributes.broker_submission_id} and
+                    {"" if self._last_id is None else f"certified_award_financial_id > {self._last_id} and"}
+                    transaction_obligated_amou is not null and
+                    transaction_obligated_amou != 0
+            order   by certified_award_financial_id
+            limit   {self.chunk_size}
+        """
+
+        award_financial_frame = pd.read_sql(sql, connections["data_broker"])
+
+        if award_financial_frame.size > 0:
+            award_financial_frame["object_class"] = award_financial_frame.apply(get_object_class_row, axis=1)
+            award_financial_frame["program_activity"] = award_financial_frame.apply(
+                get_or_create_program_activity, axis=1, submission_attributes=self.submission_attributes
+            )
+            award_financial_frame = award_financial_frame.replace({np.nan: None})
+
+            self._last_id = award_financial_frame["certified_award_financial_id"].max()
+            self._current_chunk = deque(award_financial_frame.to_dict(orient="records"))
+
+        else:
+            self._last_id = None
+            self._current_chunk = None
+
+    def __next__(self):
+        if not self._current_chunk:
+            self._retrieve_and_prepare_next_chunk()
+            if not self._current_chunk:
+                raise StopIteration
+        return self._current_chunk.popleft()
+
+
+class CertifiedAwardFinancial:
+    """ Abstract away the messy details of how we retrieve and prepare certified_award_financial rows. """
+
+    def __init__(self, submission_attributes):
+        self.submission_attributes = submission_attributes
+
+    @cached_property
+    def count(self):
+        sql = f"""
+            select  count(*)
+            from    certified_award_financial
+            where   submission_id = {self.submission_attributes.broker_submission_id} and
```

Comment:
It doesn't really matter but could define the where clause as a string and pass it to the iterator. Seems they are both the exact same except for the conditional `_last_id` check.